### DISCUSSION TITLE
Minimizing the number of files in the root directory of Meshery

### DISCUSSION THREAD

**Discussion by leecalcote** [meshery/meshery#3210] [General] [answered]
@ramrodo with the addition of MeshKit Error Utils, helm chart releaser, and CodeCov, we have 7 new files in the root directory of the meshery repo. I’m wondering if we can relocate the codecov config and coverage output. Is this possible?

**Comment by leecalcote** [comment]
@piyushsingariya can relocated this file? https://github.com/layer5io/meshery/blob/master/swagger.yaml into the helpers folder? // @navendu-pottekkat

**Reply by piyushsingariya** [reply]
Cool! I'll raise a PR.

**Comment by leecalcote** [comment]
Related issue for @mgfeller and @kumarabd to consider - https://github.com/layer5io/meshery/issues/3207

**Comment by ramrodo** [comment]
According to this:
https://docs.codecov.com/docs/codecov-yaml#can-i-name-the-file-codecovyml

> Can I name the file .codecov.yml?
> Yes. However, the file must still be located in the root, dev/, or .github/ directories

So, it is possible to move it into `.github/` directory.

**Comment by leecalcote** [comment]
@nehaLohia27 can we have this file https://github.com/meshery/meshery/blob/master/script.sh moved to the /docs directory? // @Nikhil-Ladha

**Reply by Nikhil-Ladha** [reply]
Yeah, that's very well possible.

**Comment by leecalcote** [comment, marked as answer]
@Aisuko would you be so kind as to find a new home for https://github.com/meshery/meshery/blob/master/ct.yaml inside of the `/install` folder?

**Reply by Aisuko** [reply]
For sure,  I believe we can remove this file

**Reply by Aisuko** [reply]
I will do PR for this.

**Reply by Aisuko** [reply]
~Please help to remove the branch `test/chart-release`, no permission to delete this no active branch.~

**Reply by Aisuko** [reply]
Done,  https://github.com/meshery/meshery/pull/3364

**Comment by leecalcote** [comment]
Proposal to restructure the top-level folder structure of `meshery/meshery` by major architectural component, for example:
```
/docs
/install
  docker-compose.yaml (from /)
  meshery (from /)
/mesheryctl
/provider-ui
/server
  /cmd
  /handlers
  /helpers
  /internal
  /models (from /oam)
  /meshes
  /router
/ui
```
Bold (proposed new)
